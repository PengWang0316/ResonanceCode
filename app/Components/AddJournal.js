import React, { Component } from "react";
import Util from "../apis/Util";
import Loading from "./Loading"
import JournalContent from "./JournalContent";
import ReadingSearchAndList from "./ReadingSearchAndList";
import { isLogin } from "../apis/LoginApi";
import DatabaseApi from "../apis/DatabaseApi";
import PropTypes from "prop-types";

class AddJournal extends Component {

  componentWillMount(){
    // Initial variables
    this.readings = this.props.journalData ? this.props.journalData.readingIds : {}; // keep which readings should be attached on. Format is like {readingId: null}
    this.contents = {}; // Keep content keys and content
    this.contentKeyIndex = 0; // Generate keys for different contents
    this.contentIndexs = {}; // Use to track the index of content component in the array. Delete function needs it. The format is {contentKey: index}
    this.jounalId = this.props.journalData ? this.props.journalData._id : null; // Keeping journal id for update
    this.userId = isLogin(document).userid;

    this.state={
      isWriting: false,
      journalDate: this.props.journalData ? Util.getDateString(this.props.journalData.date) : Util.getCurrentDateString(),
      isDateCorrect: true,
      isEmptyReading: this.props.journalData ? false : true,
      contentComponentArray: [], // keep content component
      addJournalContent: "overview",
      pingPongState: this.props.journalData ? this.props.journalData.ping_pong_state : "Neutral"
    };

    console.log("Journal before alter: ", this.props.journalData);
    // If journal data exsit, get content
    if(this.props.journalData){
      // this.oldContentKeys = []; // keep original content keys
      this.oldReadingIds = Object.keys(this.readings); // keep original reading ids

      // delete unnecessary properties from journal object
      let journal = this.props.journalData;
      this.journalUserId = journal.user_id;
      delete journal.date;
      delete journal.user_id;
      delete journal.ping_pong_state;
      delete journal._id;
      delete journal.readingIds;
      let keyRegExp = /-(\d+)$/; // The regular expression for subtract suffixs
      console.log("Journal after alter: ", this.props.journalData);
      Object.keys(journal).map((key)=>{
        // this.oldContentKeys.push(key); // keeping the exsit keys in an array for delete function
        // handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared)
        // Also have to save the largest key number and use it to countine
        // journalData property is like {'overview-0': 'test overview', 'overview-0-isShared': true}
        // if key is isShared do not do anything
        let matchResult = key.match(keyRegExp);
        if (matchResult){
          if (this.contentKeyIndex < matchResult[1]) this.contentKeyIndex = matchResult[1]; // keep the max index number and user can contiune add new content
          this.handleAddContentClick(journal[key], key.replace(keyRegExp, ""), key, matchResult[1], journal[`${key}-isShared`]);
          // console.log("isShared: ",journal[`${key}-isShared`]);
        }
      });
      this.contentKeyIndex++;
    }

  }

  handleChange(event, element){
    let newState={};
    newState[element]=event.target.value;
    if(element=="journalDate") newState.isDateCorrect = Util.matchDateFormat(newState[element]);
    this.setState(newState);
  }

  handleChangeCallback(contentKey, contentText){
    console.log(contentKey, contentText);
    // update content in state
    this.contents[contentKey]=contentText;
  }

  handleDeleteContentCallback(contentKey){
    // Remove content from state
    if(this.contents.hasOwnProperty(contentKey)){
      delete this.contents[contentKey];
      delete this.contents[`${contentKey}-isShared`];
    }
    // Remove content component
    delete this.contentIndexs[contentKey];
    this.setComponentToStateArray();
    // this.state.contentComponentArray.splice(this.contentIndexs[contentKey],1);
    // this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  handleSharedBoxChangeCallback(contentKey, isShared){
    this.contents[`${contentKey}-isShared`] = isShared;
  }

  handleAddContentClick(addJournalContent, newContentName, newContentKey, contentKeyIndex, isShared){
    // put component in array in order to show
    newContentKey = newContentKey ? newContentKey : `${this.state.addJournalContent}-${this.contentKeyIndex}`;
    // put the component in the track object
    this.contentIndexs[newContentKey]=<JournalContent key={contentKeyIndex ? contentKeyIndex : this.contentKeyIndex++} newContent={addJournalContent ? addJournalContent : ""} newContentName={newContentName ? newContentName : this.state.addJournalContent} newContentKey={newContentKey} handleChangeCallback={(contentKey, contentText)=>{this.handleChangeCallback(contentKey, contentText);}} handleDeleteContentCallback={(contentKey)=>{this.handleDeleteContentCallback(contentKey);}} handleSharedBoxChangeCallback={(contentKey, isShared)=>{this.handleSharedBoxChangeCallback(contentKey, isShared)}} isShared={isShared ? true : false} />;
    this.setComponentToStateArray();
    // this.setState({contentComponentArray: this.state.contentComponentArray});

    // also have to put content title to state in order to track content and update
    this.contents[newContentKey] = addJournalContent ? addJournalContent : "";
    this.contents[`${newContentKey}-isShared`] = isShared ? true : false;
  }

  setComponentToStateArray(){
    // transfor object to array in order to display
    this.state.contentComponentArray=[];
    Object.keys(this.contentIndexs).map((element)=>{
      this.state.contentComponentArray.push(this.contentIndexs[element]);
    });
    this.setState({contentComponentArray: this.state.contentComponentArray});
  }

  handleSubmit(event, isUpdate){
    event.preventDefault();
    // console.log("submit: contents:",this.contents);
    console.log("journal date:",this.state.journalDate);
    // Assemble a journal object for save
    let readingIdArray = [];
    Object.keys(this.readings).map((key)=>{readingIdArray.push(key)});
    let journal = Object.assign({_id:this.jounalId, date: new Date(this.state.journalDate), ping_pong_state: this.state.pingPongState, user_id: this.userId, readingIds: readingIdArray}, this.contents);

    if(isUpdate){
      console.log("submit: journal:", journal);
      // console.log("oldContentKeys:", this.oldContentKeys);
      console.log("oldReadingKeys:", this.oldReadingIds);
      // Assemble two arrays for deletion contents and reading
      let deleteContents = [];
      let deleteReadingIds = [];
      // this.oldContentKeys.map((element)=>{if(!this.contents.hasOwnProperty(element)) deleteContents.push(element);});
      this.oldReadingIds.map((element)=>{if(!this.readings.hasOwnProperty(element)) deleteReadingIds.push(element);});
      // journal.deleteContents = deleteContents; //todo delete deleteContents and oldContentKeys
      journal.deleteReadingIds = deleteReadingIds;
      console.log("delete contents: ",deleteContents);
      console.log("delete reading ids : ",deleteReadingIds);
      DatabaseApi.updateJournal(journal).then((result)=>{
        this.props.history.push("/reading");
      });
    }else{
      DatabaseApi.createJournal(journal).then((result)=>{
        this.props.history.push("/reading");
      });
    }
    /**/
    this.setState({isWriting: true});
    console.log("submit: journal:", journal);
  }

  /* For ReadingSearchAndList callback */
  handleAttachReadingCallback(readingId){
    this.readings[readingId] = null;
    this.setState({isEmptyReading: false});
  }

  handleDetachAttachReadingCallback(readingId){
    delete this.readings[readingId];
    this.setState({isEmptyReading: Object.keys(this.readings).length===0});
  }

  handleCancel(){
    this.props.history.push("/reading");
  }

  handleDelete(){
    console.log("Delete journal!");
  }

  render(){
    return(
      <div className="addReadingDiv">
        {this.state.isWriting && <Loading text="Creating" />}

        <div className="titleDiv">{this.props.journalData ? "Update journal for readings" : "Add a new journal for readings"}</div>
        <form className="form-horizontal" onSubmit={(event) => {this.handleSubmit(event, this.props.journalData);}}>

          <div className="text-right bottom-btn-div">
            {(!this.props.journalData || this.userId == this.journalUserId) &&
              <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect) || this.state.isEmptyReading}>{this.props.journalData ? "Update" : "Submit"}</button>
            }
            {(this.props.journalData && this.userId == this.journalUserId) &&
              <button onClick={()=>{this.handleDelete()}} type="button" className="btn btn-danger loginButton">Delete</button>
            }
            <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
          </div>

            <div className="form-group row form-div">
              <label htmlFor="journalDate" className="col-sm-1 col-form-label">Date</label>
              <div className="col-sm-4">

                <input className={this.state.isDateCorrect?"form-control":"form-control form-control-warning"} type="text" placeholder="mm/dd/yyyy" id="journalDate" value={this.state.journalDate} onChange={(event)=>{this.handleChange(event, "journalDate")}} />
                {!this.state.isDateCorrect && <span className="glyphicon glyphicon-warning-sign form-control-feedback form-control-warning-span"></span>}

              </div>
              <label htmlFor="pingPongState" className="col-sm-3 col-form-label">Ping Pong State</label>
              <div className="col-sm-4">
                <select value={this.state.pingPongState} id="pingPongState" className="form-control" onChange={(event)=>{this.handleChange(event,"pingPongState");}}>
                  <option value="Neutral">Neutral</option>
                  <option value="Reading/Call">Reading/Call</option>
                  <option value="Impact absorbing">Impact absorbing</option>
                  <option value="Impact complete">Impact complete</option>
                  <option value="Response">Response</option>
                  <option value="Outward moving">Outward moving</option>
                  <option value="Outward complete">Outward complete</option>
                </select>
              </div>
            </div>

            {/* New content goes here */}
            {this.state.contentComponentArray}

            {/* Add content button and drop list */}
            <div className="row addJournalContentDiv">

              <div onClick={()=>{this.handleAddContentClick();}} className="addJournalContentBtnDiv col-sm-6"><i className="fa fa-plus-square" /> Add one content for your journal</div>
              <div className="col-sm-6">
                <select  className="form-control" onChange={(event)=>{this.handleChange(event,"addJournalContent");}}>
                  <option value="overview">Overview</option>
                  <option value="internal_environment">Internal Environment</option>
                  <option value="relational_environment">Relational Environment</option>
                  <option value="physical_environment">Physical Environment</option>
                  <option value="creative_vector">Creative Vector</option>
                  <option value="synchronicities">Synchronicities</option>
                  <option value="dreams">Dreams</option>
                  <option value="other">Other</option>
                </select>

              </div>

            </div>


            {/*  Start reading search function */}
            <ReadingSearchAndList readings={this.props.journalData ? this.readings : null} attachReadingCallback={(readingId)=>{this.handleAttachReadingCallback(readingId);}} detachReadingCallback={(readingId)=>{this.handleDetachAttachReadingCallback(readingId);}} />

              <div className="text-right bottom-btn-div">
                {(!this.props.journalData || this.userId == this.journalUserId) &&
                  <button type="submit" className="btn btn-info loginButton" disabled={this.state.isWriting || !(this.state.journalDate.length>0) || !(this.state.isDateCorrect) || this.state.isEmptyReading}>{this.props.journalData ? "Update" : "Submit"}</button>
                }
                {(this.props.journalData && this.userId == this.journalUserId) &&
                  <button onClick={()=>{this.handleDelete()}} type="button" className="btn btn-danger loginButton">Delete</button>
                }
                <button onClick={()=>{this.handleCancel()}} type="button" className="btn btn-nomal loginButton">Cancel</button>
              </div>
          </form>

        </div>
    );
  }


}
AddJournal.propTypes={
  journalData: PropTypes.object,
  history: PropTypes.object
};
export default AddJournal;
