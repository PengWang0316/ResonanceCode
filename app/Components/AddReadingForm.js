import React, { Component } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';

import HexagramLine from './HexagramLine';
import { getCurrentDateString } from '../apis/Util';

const jQuery = require('jquery');

window.jQuery = jQuery;
require('../resources/jquery-ui.min.css');
require('../resources/jquery-ui.min');
/** The add reading form component.
 * @returns {null} No return;
*/
export class AddReadingForm extends Component {
  /*
  static propTypes = {
    readings: PropTypes.arrayOf(PropTypes.shape(PropTypes.oneOf([
      PropTypes.number,
      PropTypes.string,
      PropTypes.arrayOf(PropTypes.string),
      PropTypes.arrayOf(PropTypes.shape(PropTypes.oneOf([
        PropTypes.string,
        PropTypes.bool
      ]))),
      PropTypes.shape(PropTypes.oneOf([
        PropTypes.number,
        PropTypes.string
      ]))
    ])))
  };
  static defaultProps = {
    readings: []
  }; */
  /** Offering some states for the component.
   * @returns {object} Return a object for the initial state.
  */
  static getInitialState = () => ({
    readingName: '',
    people: '',
    date: getCurrentDateString()
  });

  state = Object.assign({}, AddReadingForm.getInitialState());

  /** Setting some state for the component.
   * @returns {null} No return.
  */
  // componentWillMount() {
  //   this.initialState();
  // }

  /** Setting up a datepicker for the Date input.
    * @returns {null} No return.
  */
  componentDidMount() {
    // Setting up datepicker
    jQuery('#date').datepicker({
      onSelect: dateText => this.setState({ date: dateText })
    });
  }

  /** Setting some state for the component.
   * @param {object} nextProps contains value for next props.
   * @returns {null} No return.
  */
  componentWillReceiveProps(nextProps) {
    if (this.props.readings.length !== 0 &&
      this.props.readings.length !== nextProps.readings.length)
      this.setState(Object.assign({}, AddReadingForm.getInitialState()));
  }

  /** Handle submit function.
   * @param {object} event is the object comes from submit a form.
   * @returns {null} No return;
  */
  handleSubmit = event => {
    event.preventDefault();
    jQuery('button[type=submit]').attr('disabled', 'disabled');
    this.props.handleSubmit(this.state);
  }

  /** Controlling the input value for the form.
   * @param {object} event is the object comes from a input's value changing.
   * @param {string} name is the input's id.
   * @returns {null} No return;
  */
  handleInputChange = event => this.setState({ [event.target.id]: event.target.value });

  /** Call the container's method when a user click a coin icon.
    * @param {int} lineNumber is the number of line.
    * @param {int} coinsPoint is the point that moment this line has based on the calculation.
    * @returns {null} No return.
  */
  handleCoinClickCallback = (lineNumber, coinsPoint) =>
    this.props.handleCoinClick(lineNumber, coinsPoint);

  /** Calling the container's cancel method when a user clicks the cancel button.
    * @returns {null} No return.
  */
  handleCancelCallback = () => this.props.handleCancel();

  /** Render the component.
   * @returns {null} No return;
  */
  render() {
    return (
      <div>
        <div className="rcTitle">Create A New Reading</div>
        <form className="form-horizontal" onSubmit={this.handleSubmit}>
          <div className="text-right bottom-btn-div">
            <button type="submit" className="btn btn-info loginButton" disabled={!(this.props.addReadingTempState.availableArr[6] && this.state.people.length > 0 && this.state.readingName.length > 0 && this.state.date.length > 0 && !this.props.addReadingTempState.isLoading)}>Submit</button>
            <button type="button" className="btn btn-normal loginButton" onClick={this.handleCancelCallback}>Cancel</button>
          </div>

          <div className="form-group row form-div">
            <label htmlFor="readingName" className="col-lg-3 col-form-label">Reading Name</label>
            <div className="col-lg-9">
              <input className="form-control" type="text" placeholder="The name for this reading" id="readingName" value={this.state.readingName} onChange={this.handleInputChange} />
            </div>
          </div>

          <div className="form-group row form-div">
            <label htmlFor="people" className="col-lg-1 col-form-label">People</label>
            <div className="col-lg-7">
              <input className="form-control" type="text" placeholder="Who is doing this with you" id="people" value={this.state.people} onChange={this.handleInputChange} />
            </div>
            <label htmlFor="date" className="col-lg-1 col-form-label">Date</label>
            <div className="col-lg-3">
              <input className="form-control" type="text" id="date" value={this.state.date} onChange={this.handleInputChange} />
            </div>
          </div>


          {/* The results of coins */}
          <div className="rcTitle coinDiv">Throw your coins and record here.</div>

          <div className="row">
            <div className="col-lg-6">
              <div className="addreading_image_title">The first image:</div>
              {this.props.addReadingTempState.availableArr[5] ? <HexagramLine lineNumber="5" side={this.props.addReadingTempState.line5.side1} middle={this.props.addReadingTempState.line5.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 6 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[4] ? <HexagramLine lineNumber="4" side={this.props.addReadingTempState.line4.side1} middle={this.props.addReadingTempState.line4.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 5 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[3] ? <HexagramLine lineNumber="3" side={this.props.addReadingTempState.line3.side1} middle={this.props.addReadingTempState.line3.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 4 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[2] ? <HexagramLine lineNumber="2" side={this.props.addReadingTempState.line2.side1} middle={this.props.addReadingTempState.line2.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 3 has not been entered.</div>}

              {this.props.addReadingTempState.availableArr[1] ? <HexagramLine lineNumber="1" side={this.props.addReadingTempState.line1.side1} middle={this.props.addReadingTempState.line1.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst /> : <div className="noAvailableDiv text-center">Line 2 has not been entered.</div>}

              <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line0.side1} middle={this.props.addReadingTempState.line0.middle1} handleCoinClick={this.handleCoinClickCallback} isFirst />
            </div>

            <div className="col-lg-6">
              <div className="addreading_image_title">The second image:</div>
              {this.props.addReadingTempState.line5.side2 === '' ? <div className="availableDiv text-center" >Line 6</div> : <HexagramLine lineNumber="5" side={this.props.addReadingTempState.line5.side2} middle={this.props.addReadingTempState.line5.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line4.side2 === '' ? <div className="availableDiv text-center">Line 5</div> : <HexagramLine lineNumber="4" side={this.props.addReadingTempState.line4.side2} middle={this.props.addReadingTempState.line4.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line3.side2 === '' ? <div className="availableDiv text-center">Line 4</div> : <HexagramLine lineNumber="3" side={this.props.addReadingTempState.line3.side2} middle={this.props.addReadingTempState.line3.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line2.side2 === '' ? <div className="availableDiv text-center">Line 3</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line2.side2} middle={this.props.addReadingTempState.line2.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line1.side2 === '' ? <div className="availableDiv text-center">Line 2</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line1.side2} middle={this.props.addReadingTempState.line1.middle2} isFirst={false} />}

              {this.props.addReadingTempState.line0.side2 === '' ? <div className="availableDiv text-center">Line 1</div> : <HexagramLine lineNumber="0" side={this.props.addReadingTempState.line0.side2} middle={this.props.addReadingTempState.line0.middle2} isFirst={false} />}
            </div>


          </div>
          <div className="change-line-div"><span>Change Lines: </span>{this.props.addReadingTempState.changeLines}</div>


          <div className="text-left bottom-btn-div">
            <button type="submit" className="btn btn-info loginButton" disabled={!(this.props.addReadingTempState.availableArr[6] && this.state.people.length > 0 && this.state.readingName.length > 0 && this.state.date.length > 0 && !this.props.addReadingTempState.isLoading)}>Submit</button>
            <button type="button" className="btn btn-normal loginButton" onClick={this.handleCancelCallback}>Cancel</button>
          </div>
        </form>
      </div>
    );
  }
}
/*
AddReadingForm.propTypes = {
  addReadingTempState: PropTypes.object.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleCoinClick: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}; */
const mapStateToProps = state => ({
  readings: state.readings
});
export default connect(mapStateToProps, null)(AddReadingForm);
