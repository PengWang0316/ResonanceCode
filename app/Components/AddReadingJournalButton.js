import React from 'react';
import { Link } from 'react-router-dom';
import jQuery from 'jquery';

import styles from '../styles/AddReadingJournalButton.module.css';

const handleAddBtnClick = () => {
  const isShowAddBtns = jQuery('#addReadingBtn').css('opacity') !== '0';
  const opacity = isShowAddBtns ? '0' : '1';
  const readingBottom = isShowAddBtns ? '25px' : '85px';
  const journalBottom = isShowAddBtns ? '25px' : '145px';
  jQuery('#addReadingBtn').css({ opacity, bottom: readingBottom });
  jQuery('#addJournalBtn').css({ opacity, bottom: journalBottom });
};

// $ will use jQuery in the index.html.
const handleAddReadingButtonClick = () => $('#addReadingModal').modal('toggle');

const AddReadingJurnalButton = () => (
  <div role="button" tabIndex="-3" onClick={handleAddBtnClick}>
    {/* this $('#addReadingModal').modal('toggle') will use jQuery from index.html */}
    <div role="button" tabIndex="-4" onClick={handleAddReadingButtonClick} id="addReadingBtn" className={`${styles.addBtn} ${styles.addReadingBtn} text-center`}>Reading</div>
    <Link to={{ pathname: '/addJournal' }}><div id="addJournalBtn" className={`${styles.addBtn} ${styles.addJournalBtn} text-center`}>Journal</div></Link>
    <div className={`${styles.addBtn}`}><i className="fa fa-plus" /></div>
  </div>
);

export default AddReadingJurnalButton;
