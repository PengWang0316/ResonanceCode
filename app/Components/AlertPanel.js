import React from 'react';

import styles from '../styles/AlertPanel.module.css';

const AlertPanel = ({
  id = '', type = 'success', children = '', style = {}
}) => (
  <div id={id} className={`alert alert-${type} mt-3 ${styles.alertPanel}`} style={style} role="alert">
    {children}
  </div>
);
export default AlertPanel;
