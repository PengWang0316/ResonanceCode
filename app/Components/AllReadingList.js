import React from 'react';
import { connect } from 'react-redux';

import Pagination from './SharedComponents/Pagination';
import LoadingAnimation from './SharedComponents/LoadingAnimation';
import { fetchAllReadingList as fetchAllReading } from '../actions/ReadingActions';
import { NUMBER_OF_READING_PER_PAGE } from '../config';
import ReadingListRow from './ReadingListRow';

/* Show the all reading in the list */
export const AllReadingList = ({
  allReadingList, readingsAmount, fetchAllReadingList, handleClick, isLoading
}) => (
  <div className="modal fade" id="readingListModal" tabIndex="-1" role="dialog" aria-labelledby="readingListModalLabel" aria-hidden="true">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="readingListModalLabel">Select a reading for this journal</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <LoadingAnimation />
          {!isLoading &&
            <div className="mb-3">
              {allReadingList.map(reading => <ReadingListRow key={`${reading.name}_${reading.date}`} reading={reading} handleClick={handleClick} />)}
            </div>
          }

          {/* Pagination componentWillMount
            * Keeping readingsAmount, numberPerpage, and fetchContent out of AllReadingListPagination component can help it raising reusability.
          */}
          {readingsAmount !== null && <Pagination
            amount={readingsAmount}
            fetchContent={fetchAllReadingList}
            numberPerpage={NUMBER_OF_READING_PER_PAGE}
          />}

        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);
const mapStateToProps = state => ({
  allReadingList: state.allReadingList,
  readingsAmount: state.readingsAmount,
  isLoading: state.isLoading
});
const mapDispatchToProps = dispatch => ({
  fetchAllReadingList: pageNumber => dispatch(fetchAllReading(pageNumber))
});
// export AllReadingList;
export default connect(mapStateToProps, mapDispatchToProps)(AllReadingList);
