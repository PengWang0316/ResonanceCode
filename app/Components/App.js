
import React from "react";
// import { CSSTransitionGroup } from 'react-transition-group';
// import Popular from "./Popular";
// import Home from "./Home";
// import Battle from "./Battle";
// import Results from "./Results";
import Nav from "./Nav";
import Login from "./Login";
import Reading from "./Reading";
import DetailedReading from "./DetailedReading";
import AddReading from "./AddReading";
import AddJournal from "./AddJournal";
import ShowJournal from "./ShowJournal";
import JournalList from "./JournalList";
import SignUp from "./SignUp";
import ReadingSearch from "./search/ReadingSearch";
import HexagramsSearch from "./search/HexagramsSearch";
import Hexagrams from "./administration/Hexagrams";
import {BrowserRouter as Router, Route, Switch} from "react-router-dom";

const App=()=>{
  return(
    <Router>
      <div className="container">
        <Nav />
        <main>
          {/*
            <CSSTransitionGroup
            transitionName="csstransition-fade"
            transitionEnterTimeout={300}
            transitionLeaveTimeout={300}
          >
            */}
          <Switch>
            <Route exact path="/" component={Login} />
            <Route path="/reading" component={Reading} />
            <Route path="/detailedreading" component={DetailedReading} />
            <Route path="/addreading" component={AddReading} />
            <Route path="/addJournal" component={AddJournal} />
            <Route path="/showJournal" component={ShowJournal} />
            <Route path="/journalList" component={JournalList} />
            <Route path="/readingSearch" component={ReadingSearch} />
            <Route path="/hexagramsSearch" component={HexagramsSearch} />
            <Route path="/signup" component={SignUp} />
            <Route path="/unattachedJournals" component={JournalList} />
            {/*For administrator*/}
            <Route path="/hexagrams" component={Hexagrams} />
            <Route render={()=>{return(<p>Not Fount!</p>)}} />
          </Switch>
          {/*</CSSTransitionGroup>*/}

        </main>
      </div>
    </Router>
  );
};

export default App;
