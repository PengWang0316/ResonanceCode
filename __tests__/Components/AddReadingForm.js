import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { AddReadingForm } from '../../app/Components/AddReadingForm';
// import Unit from '../../app/apis/Util';

jest.mock('../../app/Components/HexagramLine', () => 'HexagramLine');
// Unit.getCurrentDateString = jest.fn(() => '01/13/2018');
jest.mock('../../app/apis/Util', () => ({ getCurrentDateString: jest.fn(() => '01/13/2018') }));
jest.mock('../../app/resources/jquery-ui.min');

describe('Test AddReadingForm', () => {
  let props;
  let shallowAddReadingForm;
  // let mountAddReadingForm;
  const addReadingFormShallow = () => {
    if (!shallowAddReadingForm) shallowAddReadingForm = shallow(<AddReadingForm {... props} />);
    return shallowAddReadingForm;
  };

  // const addReadingFormMount = () => {
  //   if (!mountAddReadingForm) mountAddReadingForm = mount(<AddReadingForm {... props} />);
  //   return mountAddReadingForm;
  // };

  beforeEach(() => {
    props = {
      readings: [],
      addReadingTempState: {
        line0: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        line1: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        line2: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        line3: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        line4: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        line5: {
          side1: '', middle1: '', side2: '', middle2: ''
        },
        changeLines: '',
        availableArr: [true, false, false, false, false, false, false],
        /* readingName: "",
      date: Util.getCurrentDateString(),
      people: "",
      isWriting: false, */
        imageArrays: { img1: [], img2: [] },
        changeLinesNumberArray: []
      },
      handleCancel: jest.fn(),
      handleCoinClick: jest.fn(),
      handleSubmit: jest.fn()
    };
    shallowAddReadingForm = undefined;
  });

  test('Has a form element', () => {
    const forms = addReadingFormShallow().find('form');
    expect(forms.length).toBeGreaterThan(0);
  });

  test('componentWillReceiveProps', () => {
    const addReadingForm = addReadingFormShallow();
    addReadingForm.find('form').simulate('submit', { preventDefault() {} });
    addReadingForm.setState({ // Manually set up state to values that are differetn with inintail state.
      readingName: 'aa',
      people: 'aa',
      date: null,
      isDateCorrect: false
    });
    addReadingForm.setProps({ readings: [1, 2] });
    const newState = addReadingForm.state();
    expect(newState.readingName).toBe('');
    expect(newState.people).toBe('');
    expect(newState.isDateCorrect).toBe(true);
  });

  test('handleSubmit', () => {
    const addReadingForm = addReadingFormShallow();
    addReadingForm.find('form').simulate('submit', { preventDefault() {} });
    expect(addReadingForm.instance().props.handleSubmit).toHaveBeenCalled();
  });

  test('Snapshot test', () => {
    const tree = renderer.create(<AddReadingForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
