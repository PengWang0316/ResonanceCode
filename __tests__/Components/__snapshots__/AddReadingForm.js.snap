// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddReadingForm Snapshot test 1`] = `
<div>
  <div
    className="title"
  >
    Create A New Reading
  </div>
  <form
    className="form-horizontal"
    onSubmit={[Function]}
  >
    <div
      className="text-right mt-4"
    >
      <button
        className="btn btn-info submitButton"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
      <button
        className="btn btn-normal submitButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="form-group row formDiv"
    >
      <label
        className="col-lg-3 col-form-label"
        htmlFor="readingName"
      >
        Reading Name
      </label>
      <div
        className="col-lg-9"
      >
        <input
          className="form-control"
          id="readingName"
          onChange={[Function]}
          placeholder="The name for this reading"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group row formDiv"
    >
      <label
        className="col-lg-1 col-form-label"
        htmlFor="people"
      >
        People
      </label>
      <div
        className="col-lg-7"
      >
        <input
          className="form-control"
          id="people"
          onChange={[Function]}
          placeholder="Who is doing this with you"
          type="text"
          value=""
        />
      </div>
      <label
        className="col-lg-1 col-form-label"
        htmlFor="date"
      >
        Date
      </label>
      <div
        className="col-lg-3"
      >
        <input
          className="form-control"
          id="date"
          onChange={[Function]}
          type="text"
          value="01/13/2018"
        />
      </div>
    </div>
    <div
      className="title coinDiv"
    >
      Throw your coins and record here.
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-6"
      >
        <div
          className="addreadingImageTitle"
        >
          The first image:
        </div>
        <div
          className="text-center noAvailableDiv"
        >
          Line 6 has not been entered.
        </div>
        <div
          className="text-center noAvailableDiv"
        >
          Line 5 has not been entered.
        </div>
        <div
          className="text-center noAvailableDiv"
        >
          Line 4 has not been entered.
        </div>
        <div
          className="text-center noAvailableDiv"
        >
          Line 3 has not been entered.
        </div>
        <div
          className="text-center noAvailableDiv"
        >
          Line 2 has not been entered.
        </div>
        <HexagramLine
          handleCoinClick={[Function]}
          isFirst={true}
          lineNumber="0"
          middle=""
          side=""
        />
      </div>
      <div
        className="col-lg-6"
      >
        <div
          className="addreadingImageTitle"
        >
          The second image:
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 6
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 5
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 4
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 3
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 2
        </div>
        <div
          className="text-center availableDiv"
        >
          Line 1
        </div>
      </div>
    </div>
    <div
      className="changeLineDiv"
    >
      <span>
        Change Lines: 
      </span>
      
    </div>
    <div
      className="text-left mt-4"
    >
      <button
        className="btn btn-info submitButton"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
      <button
        className="btn btn-normal submitButton"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
